### PR TITLE
Keep query string in Statement object

### PR DESCRIPTION
**Pull request by mbutrovich**
Statement now takes ownership of the query string rather than allowing it to be discarded after parsing. Also refactored some classes (Statement, Portal, BinderSherpa) to have .cpp files to adhere to codebase best practices.

The main change is the change to Statement's constructor and the addition of GetQueryText on that class. The rest is just refactoring.

### CODE REVIEW

File: src/network/postgres/statement.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#include "type/type_id.h"
+
+namespace terrier::network {
+
+Statement::Statement(std::string &&query_text, std::unique_ptr<parser::ParseResult> &&parse_result)
+    : Statement(std::move(query_text), std::move(parse_result), {}) {}
+
+Statement::Statement(std::string &&query_text, std::unique_ptr<parser::ParseResult> &&parse_result,
+                     std::vector<type::TypeId> &&param_types)
+    : parse_result_(std::move(parse_result)), param_types_(std::move(param_types)) {
```
Comment (malin1993ml) [inline src/network/postgres/statement.cpp:21]: Seems that you need to move `query_text` into `query_text_` here?

File: src/include/binder/binder_sherpa.h [hunk 2]
```diff
@@ -31,41 +32,32 @@ class BinderSherpa {
...
    */
-  common::ManagedPointer<parser::ParseResult> GetParseResult() const { return parse_result_; }
+  common::ManagedPointer<parser::ParseResult> GetParseResult() const;
 
   /**
    * @return parameters for the query being bound
    * @warning can be nullptr if there are no parameters
    */
-  common::ManagedPointer<std::vector<type::TransientValue>> GetParameters() const { return parameters_; }
+  common::ManagedPointer<std::vector<type::TransientValue>> GetParameters() const;
```
Comment (malin1993ml) [inline src/include/binder/binder_sherpa.h:47]: If I understand gcc correctly, the compiler cannot inline these functions if you put them in the cpp file. Just curious is there a reason to move these one-liner functions into the cpp file?

File: src/network/postgres/statement.cpp (hunk 1 above)
Comment (mbutrovich) [reply src/network/postgres/statement.cpp:21]: Whoops. Good catch.

File: src/include/binder/binder_sherpa.h (hunk 2 above)
Comment (mbutrovich) [reply src/include/binder/binder_sherpa.h:47]: You're right. For the one-liners I should just leave them in the headers.

**Pull request merged**